### PR TITLE
K8s dependency bump

### PR DESCRIPTION
**Pull request by dprotaso**
- bumps k8s deps to v0.19.7
- drops the `require` block from the k8s deps 
  - these should just be inherited from `knative.dev/pkg` and `github.com/tektoncd/pipeline`.

After running codegen I noticed a test failing and it seems like the order of something was changed - unsure why see: https://github.com/tektoncd/triggers/commit/5e393f2c0c4c5ba511f82688c2a2fe0d98fe9758

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/google/uuid v1.2.0
 	github.com/gorilla/mux v1.7.4
 	github.com/spf13/cobra v1.0.0
-	github.com/tektoncd/pipeline v0.20.0
+	github.com/tektoncd/pipeline v0.20.1-0.20210128163741-1eca890e74c9
```
Comment (MarcelMue) [inline go.mod:15]: Why is this an untagged version of pipeline?

File: go.sum [hunk 2]
```diff
@@ -1314,53 +1389,77 @@ honnef.co/go/tools v0.0.1-2020.1.4 h1:UoveltGrhghAA7ePc+e+QYDHXrBps2PqFZiHkGR/xK
...
-k8s.io/apiextensions-apiserver v0.18.12/go.mod h1:nihADkPed1L37Vxpz2/BrtxO9mCtINH23aNtUe/CRLo=
-k8s.io/apimachinery v0.18.12 h1:bLFXU4IxOu06F6Z6PV7eqtapXFb1G2q0ni0XBNFtJH8=
-k8s.io/apimachinery v0.18.12/go.mod h1:PF5taHbXgTEJLU+xMypMmYTXTWPJ5LaW8bfsisxnEXk=
+k8s.io/api v0.18.8/go.mod h1:d/CXqwWv+Z2XEG1LgceeDmHQwpUJhROPx16SlxJgERY=
+k8s.io/api v0.19.7 h1:MpHhls03C2pyzoYcpbe4QqYiiZjdvW+tuWq6TbjV14Y=
+k8s.io/api v0.19.7/go.mod h1:KTryDUT3l6Mtv7K2J2486PNL9DBns3wOYTkGR+iz63Y=
+k8s.io/apiextensions-apiserver v0.19.7 h1:aV9DANMSCCYBEMbtoT/5oesrtcciQrjy9yqWVtZZL5A=
+k8s.io/apiextensions-apiserver v0.19.7/go.mod h1:XJNNtjISNNePDEUClHt/igzMpQcmjVVh88QH+PKztPU=
+k8s.io/apimachinery v0.0.0-20190703205208-4cfb76a8bf76/go.mod h1:M2fZgZL9DbLfeJaPBCDqSqNsdsmLN+V29knYJnIXlMA=
+k8s.io/apimachinery v0.18.8/go.mod h1:6sQd+iHEqmOtALqOFjSWp2KZ9F0wlU/nWm0ZgsYWMig=
```
Comment (MarcelMue) [inline go.sum:1398]: Maybe we should not get rid of the replace in `go.mod` because it seems like we still import 0.18.8 like this.

File: go.mod (hunk 1 above)
Comment (dprotaso) [reply go.mod:15]: I bumped it to the latest so we'd get the proper k8s deps in the `require` block

https://github.com/tektoncd/pipeline/pull/3724

File: go.sum (hunk 2 above)
Comment (dprotaso) [reply go.sum:1398]: the `go.sum` file is just a list of modules and hash we've seen

what's being used is defined in `go.mod` you can see what's being used in the `vendor/modules.txt` file

Comment (MarcelMue) [reply go.sum:1398]: This is not entirely accurate AFAIK - after running go mod tidy the `go.sum` contains direct dependencies but also transitive dependencies. In this case `v0.18.8` is most likely a transitive dependency (we can probably find out why with go mod graph).

You are correct however, it should be using the newer version. These transitive dependencies are only a problem if we run security scanners such as nancy (see here for example: https://github.com/sonatype-nexus-community/nancy/issues/152#issuecomment-675716615 ).

TL;DR you are correct it should not matter as we do not use such scanners as far as I can find.

Comment (dprotaso) [reply go.sum:1398]: oh interesting - I think it's https://github.com/google/go-containerregistry/blob/main/pkg/authn/k8schain/go.mod#L6

Comment (dprotaso) [reply go.sum:1398]: let me fix that

**Review by dibyom — COMMENTED** [review]
Thanks @dprotaso 

One question: Does the bump the min version of k8s required to install Triggers to v1.19?

File: go.sum (hunk 2 above)
Comment (dprotaso) [reply go.sum:1398]: oddly I don't see v0.18.x in the vendored folder modules.txt - which I believe takes precedent over the module cache

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go [hunk 3]
```diff
@@ -1120,16 +1120,18 @@ func TestReconcile_Delete(t *testing.T) {
 		key:  fmt.Sprintf("%s/%s", namespace, "el-2"),
 		startResources: test.Resources{
 			Namespaces: []*corev1.Namespace{namespaceResource},
-			EventListeners: []*v1alpha1.EventListener{makeEL(), makeEL(withFinalizer, withDeletionTimestamp, func(el *v1alpha1.EventListener) {
-				el.Name = "el-2" // TODO: makeEL take name, ns as args
-			})},
+			EventListeners: []*v1alpha1.EventListener{
```
[the remark this answers is not in the document]
Comment (dprotaso) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go:1123]: yup

File: hack/update-deps.sh [hunk 4]
```diff
@@ -21,7 +21,8 @@ source $(git rev-parse --show-toplevel)/vendor/github.com/tektoncd/plumbing/scri
 
 cd ${REPO_ROOT_DIR}
 
-VERSION="release-0.20"
+# Update when you want to pin knative.dev/pkg
+VERSION="master"
```
[the remark this answers is not in the document]
Comment (dprotaso) [reply hack/update-deps.sh:25]: I actually forgot to update that property in pipeline repo when I did a bump to master

see: https://github.com/tektoncd/pipeline/pull/3724

**Comment by dprotaso** [comment]
> One question: Does the bump the min version of k8s required to install Triggers to v1.19?

No K8s policy supports N, N-1 - so the min version would be K8s 1.18

That's the next target for Knative & `knative.dev/pkg`

File: go.sum (hunk 2 above)
Comment (dprotaso) [reply go.sum:1398]: I'm sorta confused now 

```
$ go list -m all  | grep v0.18
github.com/aws/aws-sdk-go-v2 v0.18.0
github.com/lightstep/lightstep-tracer-go v0.18.1
k8s.io/cloud-provider v0.18.8
k8s.io/csi-translation-lib v0.18.8
k8s.io/legacy-cloud-providers v0.18.8
```

Is says there's a dep

but when you go to each binary in this repo it says it's not used 

```
dmad:controller dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:controller dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:controller dprotasowski$ cd ../binding-eval/
dmad:binding-eval dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:binding-eval dprotasowski$ cd ../triggerrun/
dmad:triggerrun dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:triggerrun dprotasowski$ cd ../interceptors/
dmad:interceptors dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:interceptors dprotasowski$ cd ../eventlistenersink/
dmad:eventlistenersink dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:eventlistenersink dprotasowski$ cd ../webhook/
dmad:webhook dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:webhook dprotasowski$ export GO111MODULE=on
dmad:webhook dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
```

Comment (dprotaso) [reply go.sum:1398]: I see where it's coming from

ie the graph
```
github.com/google/go-containerregistry/pkg/authn/k8schain@v0.0.0-20210127165842-51f01e739161 github.com/google/go-containerregistry@v0.3.0
github.com/google/go-containerregistry/pkg/authn/k8schain@v0.0.0-20210127165842-51f01e739161 github.com/vdemeester/k8s-pkg-credentialprovider@v1.18.1-0.20201019120933-f1d16962a4db
```

Comment (dprotaso) [reply go.sum:1398]: I'm guessing the modules have dependencies but the specific packages aren't being used so it doesn't factor into the version selected?

Comment (MarcelMue) [reply go.sum:1398]: Sorry for creating this rabbit hole for you - yes 0.18.X is not actually used in the compiled code. My initial question was just if we in tekton / this project care if a transitive dependency import was still pointing elsewhere. Actual compiled code will all use 0.20.X

This section also explains how always the higher version is taken in these cases: https://golang.org/ref/mod#major-version-suffixes

So yeah, sorry again for the confusion this caused :)

Comment (dprotaso) [reply go.sum:1398]: No worries - it forces us to clean up a bunch of things upstream

**Comment by dibyom** [comment]
> No K8s policy supports N, N-1 - so the min version would be K8s 1.18

Ah ok, so the min version of k8s needed right now is v1.17 https://github.com/tektoncd/triggers/blob/master/docs/install.md#pre-requisites

mind updating that page and squashing up the commits? Otherwise good to go!

**Comment by dprotaso** [comment]
Squashed & doc updated. I fixed upstream so there shouldn't be any dep pointing to k8s 1.18 modules

**Review by MarcelMue — APPROVED** [review]
/lgtm

File: hack/update-deps.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (dprotaso) [reply hack/update-deps.sh:25]: it's set to master in `tekton/pipeline` now
https://github.com/tektoncd/pipeline/blob/98507859fa961f2152b65ad911defc2f0a2ef804/hack/update-deps.sh#L25

**Review by dibyom — APPROVED** [review]
